libs/vm/include/vm/variant.hpp

```diff
@@ -475,37 +475,23 @@ struct Variant
   }
 };
 
-struct TemplateParameter1 : public Variant
+template <std::size_t ID = 0>
+struct VariantT : public Variant
 {
   using Variant::Variant;
+  VariantT(Variant const &other)
+    : Variant{other} {};
+  VariantT(Variant &&other)
+    : Variant{std::move(other)} {};
```

Comment:
redundant semicolon; use std:;forward

Reply:
The way how the code is writtenright now, switching here to the `std::forward<...>` does not make sense unless this is the perfect forwarding.
In order of beying perfect-forwardable, the method/function must templated with one of template parameters beying type of paremeter which is intended to be forwareded.
Otherwise it is just move, because it is explicitly declared that this is cnstructor *rvalue* type .

The constructor would need to be templated, as follow:
```
template<typename T>
VariantT(T &&other) : Variant{std::forward<T>(other)}
{}
```

However, IF we make the constructor templated with perferct forwarding, as drafted in above example, we do not need to have 2 cosntructors (we can drop copy constructor since it will be covered with perfect forwarding constructor).